charlotte_core/src/arch/x86_64/memory/mod.rs

```diff
@@ -0,0 +1,55 @@
+pub mod page_map;
+
+use core::arch::x86_64::__cpuid_count;
+
+use crate::memory::address::{PhysicalAddress, VirtualAddress};
+use crate::memory::pmm::Error as PmmError;
+use spin::lazy::Lazy;
+
+/// The number of significant binary digits in a physical address
+pub static PADDR_SIGBITS: Lazy<u8> = Lazy::new(|| {
+    let cpuid_result = unsafe { __cpuid_count(0x80000008, 0) };
+    (cpuid_result.eax & 0xFF) as u8
+});
+
+/// This is the number of significant binary digits in a virtual (linear) address
```

Comment:
Please use the `#[allow(unused)]` directive to ignore warnings in our DevOps. You can lint your code manually by running `./tools/lint.sh`. Other places have bits of code that is unused for now, ignoring them would be great. Disabling this lint is another solution. Either way, we should keep our lints clean of warnings.